### PR TITLE
feat(parser): report nearby tokens when parse failed

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Partially refine #7863

```sql
dev=> create table (v1 int, v2 int);
ERROR:  QueryError: sql parser error: Near "create table (", Expected identifier, found: (
```

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by TennyZhuang** [comment]
Seems a git mistake (

**Review by yuhao-su — APPROVED** [review]
LGTM

File: src/sqlparser/src/parser.rs [hunk 1]
```diff
@@ -1475,7 +1475,24 @@ impl Parser {
...
+        impl<'a> fmt::Display for TokensDisplay<'a> {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                for token in self.0 {
+                    write!(f, "{}", token)?;
+                }
+                Ok(())
+            }
+        }
+        parser_err!(format!(
+            "Near \"{}\", Expected {}, found: {}",
```
Comment (TennyZhuang) [inline src/sqlparser/src/parser.rs:1493]:
```suggestion
            "Near \"{}\", expected {}, found: {}",
```

File: src/sqlparser/src/parser.rs [hunk 2]
```diff
@@ -1475,7 +1475,23 @@ impl Parser {
...
+                    write!(f, "{}", token)?;
+                }
+                Ok(())
+            }
+        }
+        parser_err!(format!(
+            "Near \"{}\", expected {}, found: {}",
+            TokensDisplay(near_tokens),
+            expected,
+            found
```
Comment (xxchan) [inline src/sqlparser/src/parser.rs:1493]: What about
```suggestion
        parser_err!(format!(
            "expected {}, found: {}\nNear \"{}\"",
            expected,
            found,
            TokensDisplay(near_tokens)
```

**Review by chenzl25 — APPROVED** [review]

**Review by fuyufjh — APPROVED** [review]
May add a simple SLT test case.

**Pull request merged**